src/pages/signin/PasswordForm.js

```diff
@@ -52,202 +52,210 @@ const defaultProps = {
+            input2FA.current.clear();
         }
-        this.setState({formError: {}});
+        setFormError({});
         Session.resetPassword();
-    }
+    };
 
     /**
      * Clears local and Onyx sign in states
      */
-    clearSignInData() {
-        this.setState({twoFactorAuthCode: '', formError: {}});
+    const clearSignInData = () => {
+        setTwoFactorAuthCode('');
+        setFormError({});
         Session.clearSignInData();
-    }
+    };
 
     /**
      * Check that all the form fields are valid, then trigger the submit callback
      */
-    validateAndSubmitForm() {
-        const password = this.state.password.trim();
-        const twoFactorCode = this.state.twoFactorAuthCode.trim();
-        const requiresTwoFactorAuth = this.props.account.requiresTwoFactorAuth;
+    const validateAndSubmitForm = useCallback(() => {
+        const passwordTrimmed = password.trim();
+        const twoFactorCodeTrimmed = twoFactorAuthCode.trim();
+        const requiresTwoFactorAuth = props.account.requiresTwoFactorAuth;
+
+        if (!passwordTrimmed) {
+            setFormError({password: 'passwordForm.pleaseFillPassword'});
+            return;
+        }
+
+        if (!ValidationUtils.isValidPassword(passwordTrimmed)) {
+            setFormError({password: 'passwordForm.error.incorrectPassword'});
+            return;
+        }
 
-        if (!password) {
-            this.setState({formError: {password: 'passwordForm.pleaseFillPassword'}});
+        if (requiresTwoFactorAuth && !twoFactorCodeTrimmed) {
+            setFormError({twoFactorAuthCode: 'passwordForm.pleaseFillTwoFactorAuth'});
             return;
         }
 
-        if (!ValidationUtils.isValidPassword(password)) {
-            this.setState({formError: {password: 'passwordForm.error.incorrectPassword'}});
+        if (requiresTwoFactorAuth && !ValidationUtils.isValidTwoFactorCode(twoFactorCodeTrimmed)) {
+            setFormError({twoFactorAuthCode: 'passwordForm.error.incorrect2fa'});
             return;
         }
 
-        if (requiresTwoFactorAuth && !twoFactorCode) {
-            this.setState({formError: {twoFactorAuthCode: 'passwordForm.pleaseFillTwoFactorAuth'}});
+        setFormError({});
+
+        Session.signIn(passwordTrimmed, '', twoFactorCodeTrimmed, props.preferredLocale);
+    }, [password, twoFactorAuthCode, props.account.requiresTwoFactorAuth, props.preferredLocale]);
+
+    useEffect(() => {
+        if (!canFocusInputOnScreenFocus() || !inputPasswordRef.current || !props.isVisible) {
+            return;
+        }
+        inputPasswordRef.current.focus();
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- we don't want this effect to run again
+    }, []);
+
+    useEffect(() => {
+        if (props.isVisible) {
+            inputPasswordRef.current.focus();
+        }
+        if (!props.isVisible && password) {
+            clearPassword();
+        }
+    }, [props.isVisible, password]);
```

Comment:
@fedirjh I figured out the way to solve this. We can just return the `clearPassword` method as cleanUp and it will execute once this component hides so https://github.com/Expensify/App/pull/6723 will also be solved.
<img width="492" alt="Screenshot 2023-06-14 at 4 34 03 PM" src="https://github.com/Expensify/App/assets/77237602/e70080d2-2709-4ddd-8d88-e01416d947af">

Reply:
Let me know if you think this is good